### ISSUE TITLE
About TransMIL TCGA Test Accuracy 85.24%? 

### ISSUE THREAD

**Issue by lihua8848** [HHHedo/IBMIL#8]
Thank you for your interesting work, but I have questions about the results of TransMIL's comparative experiment in TCGA in the article. The data set is the data set of the [DSMIL](https://github.com/binli123/dsmil-wsi)used in your article, and the training set and test set are the same as in the article.
<img width="422" alt="f20f962b616cb3720fb420fa1ec66c6" src="https://github.com/HHHedo/IBMIL/assets/54617440/7b7af0a4-f3e8-4d17-954c-0018ff09de8f">
The official code of TransMIL is [here](https://github.com/szc19990412/TransMIL), the parameters used are also official parameters, and the results I ran were **93.3%** accurate.
<img width="336" alt="image" src="https://github.com/HHHedo/IBMIL/assets/54617440/f9e42165-702b-4f3b-a10a-ddf52ce7129f">
How did **85.24%** of the article come from
<img width="452" alt="image" src="https://github.com/HHHedo/IBMIL/assets/54617440/f94c3d0b-541d-4b1c-8e94-ecdfc2b261d7">

In addition, I also noticed the article SSLP: Spatial Guided Self-supervised Learning on Pathological Images of your team MICCAI2021 a few years ago, which said that the code would be made public, but it has not been made public at present

Can you explain that? Thank you！

**Comment by lihua8848** [comment]
<img width="412" alt="image" src="https://github.com/HHHedo/IBMIL/assets/54617440/17f6e325-1e8a-44a3-9d34-f1ab52cd985b">

The training and testing dataset is [here](https://uwmadison.box.com/shared/static/tze4yqclajbdzjwxyb8b1umfwk9vcdwq.zip) in DSMIL official code, and you should split it to training and testing dataset like TransMIL format.
<img width="604" alt="image" src="https://github.com/HHHedo/IBMIL/assets/54617440/46fe3a42-ca9f-45ae-8ea0-98bab95d8fea">

This is the csv we made
[tcga_NSCLC.CSV](https://github.com/HHHedo/IBMIL/files/12913295/tcga_NSCLC.CSV)

**Comment by HHHedo** [comment]
> Thank you for your interesting work, but I have questions about the results of TransMIL's comparative experiment in TCGA in the article. The data set is the data set of the [DSMIL](https://github.com/binli123/dsmil-wsi)used in your article, and the training set and test set are the same as in the article. <img alt="f20f962b616cb3720fb420fa1ec66c6" width="422" src="https://user-images.githubusercontent.com/54617440/275420050-7b7af0a4-f3e8-4d17-954c-0018ff09de8f.png"> The official code of TransMIL is [here](https://github.com/szc19990412/TransMIL), the parameters used are also official parameters, and the results I ran were **93.3%** accurate. <img alt="image" width="336" src="https://user-images.githubusercontent.com/54617440/275419656-f9e42165-702b-4f3b-a10a-ddf52ce7129f.png"> How did **85.24%** of the article come from <img alt="image" width="452" src="https://user-images.githubusercontent.com/54617440/275418477-f94c3d0b-541d-4b1c-8e94-ecdfc2b261d7.png">
> 
> In addition, I also noticed the article SSLP: Spatial Guided Self-supervised Learning on Pathological Images of your team MICCAI2021 a few years ago, which said that the code would be made public, but it has not been made public at present
> 
> Can you explain that? Thank you！

Hi，
We use the patches released by dsmil and extract the feature following [dsmil](https://github.com/binli123/dsmil-wsi/blob/master/compute_feats.py). So we guess the different results mainly come from the different splits.
We do not split the dataset like the TransMIL format, however we keep the same splits for all the feature extractor and aggregators, which is a fair comparison.

For SSLP, we have extended it into a journal paper [SGCL](https://github.com/HHHedo/SGCL) (accepted by MedIA), and the code was also released.

**Comment by HHHedo** [comment]
BTW, we reproduce all the compared methods instead of directly citing their results, so there is no worry about a fair comparison like that [issue](https://github.com/szc19990412/TransMIL/issues/19).
And you may also try interventional training on your split.
Is there anything we should further explain?

**Comment by lihua8848** [comment]
> > Thank you for your interesting work, but I have questions about the results of TransMIL's comparative experiment in TCGA in the article. The data set is the data set of the [DSMIL](https://github.com/binli123/dsmil-wsi)used in your article, and the training set and test set are the same as in the article. <img alt="f20f962b616cb3720fb420fa1ec66c6" width="422" src="https://user-images.githubusercontent.com/54617440/275420050-7b7af0a4-f3e8-4d17-954c-0018ff09de8f.png"> The official code of TransMIL is [here](https://github.com/szc19990412/TransMIL), the parameters used are also official parameters, and the results I ran were **93.3%** accurate. <img alt="image" width="336" src="https://user-images.githubusercontent.com/54617440/275419656-f9e42165-702b-4f3b-a10a-ddf52ce7129f.png"> How did **85.24%** of the article come from <img alt="image" width="452" src="https://user-images.githubusercontent.com/54617440/275418477-f94c3d0b-541d-4b1c-8e94-ecdfc2b261d7.png">
> > In addition, I also noticed the article SSLP: Spatial Guided Self-supervised Learning on Pathological Images of your team MICCAI2021 a few years ago, which said that the code would be made public, but it has not been made public at present
> > Can you explain that? Thank you！
> 
> Hi， We use the patches released by dsmil and extract the feature following [dsmil](https://github.com/binli123/dsmil-wsi/blob/master/compute_feats.py). So we guess the different results mainly come from the different splits. We do not split the dataset like the TransMIL format, however we keep the same splits for all the feature extractor and aggregators, which is a fair comparison.
> 
> For SSLP, we have extended it into a journal paper [SGCL](https://github.com/HHHedo/SGCL) (accepted by MedIA), and the code was also released.


Thank you for your patient answer, it is really helpful. But i still have some questions. In our experiment, we used the following pre-processed features, and I found this [issue](https://github.com/binli123/dsmil-wsi/issues/72), he used self supervised training (simclr) to extract features, so I think this is the reason for such a big difference in our results.

<img width="648" alt="image" src="https://github.com/HHHedo/IBMIL/assets/54617440/91461378-88e9-4366-b0e8-c36736cd79da">
---

However, according to your answer, do you use dsmil to cut patches from the original image and then use resnet18 to extract the feature?
1. Download the TCGA-NSCLC dataset
2. Cut the patches using dsmil
`  $ python deepzoom_tiler.py -m 0 -b 20`
3. Use resnet18 to extract the feature
` $ python compute_feats.py --dataset=TCGA-lung-single`

I would like to know the specific parameters about cutting patches and using resnet18 to extract features? Thank you again for your work and answers.

**Comment by HHHedo** [comment]
> issue

As mentioned before, we use the tcga-nsclc patches released by [dsmil](https://github.com/binli123/dsmil-wsi/issues/16). Please refer to their great work.
